src/Features/Core/Portable/ConvertToInterpolatedString/AbstractConvertPlaceholderToInterpolatedStringRefactoringProvider.cs

```diff
@@ -35,29 +38,32 @@ internal abstract class AbstractConvertPlaceholderToInterpolatedStringRefactorin
     where TArgumentListExpressionSyntax : SyntaxNode
     where TInterpolationSyntax : SyntaxNode
 {
-    protected abstract TExpressionSyntax ParseExpression(string text);
+    private readonly record struct InvocationData(
+        TInvocationExpressionSyntax Invocation,
+        TArgumentSyntax PlaceholderArgument,
+        IMethodSymbol InvocationSymbol,
+        TInterpolatedStringExpressionSyntax InterpolatedString,
+        bool ShouldReplaceInvocation);
 
-    public override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
-    {
-        var (document, span, cancellationToken) = context;
-        var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+    protected abstract TExpressionSyntax ParseExpression(string text);
 
-        var stringType = semanticModel.Compilation.GetSpecialType(SpecialType.System_String);
-        if (stringType.IsErrorType())
-            return;
+    protected override ImmutableArray<FixAllScope> SupportedFixAllScopes { get; } = AllFixAllScopes;
 
+    private InvocationData? AnalyzeInvocation(
+        Document document,
+        SemanticModel semanticModel,
+        TInvocationExpressionSyntax invocation,
+        TArgumentSyntax placeholderArgument,
+        CancellationToken cancellationToken)
+    {
```

Comment:
core analysis routine was extracted, so it can be used in teh normal refactoring and in fix all.